### DISCUSSION TITLE
Feature Request: Allow concurrent queries with different connections in duckdb-node

### DISCUSSION THREAD

**Discussion by tronis470** [duckdb/duckdb#9405] [Ideas]
See this issue for more details: https://github.com/duckdb/duckdb-node/issues/16

In duckdb-node, queries running against different connections on the same database handle block each other. You can only run 1 query per handle at a time, regardless of using multiple connections. This is in contrast to other interfaces for DuckDB like Python, which do support concurrent queries across connections, and prevents NodeJS users from maximizing resource usage with concurrency and connection pooling.

Without this feature, the ability to build a scalable NodeJS service around DuckDB is fairly limited, as concurrent requests are constantly blocking each other, regardless of the size of the queries and size of the machine servicing those requests.

**Comment by kuatroka** [comment]
Very important indeed. Without this, duckdb-node can’t really be uses as a db for a web app
